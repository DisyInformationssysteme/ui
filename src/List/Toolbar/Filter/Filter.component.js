import React, { PropTypes } from 'react';
import { Button } from 'react-bootstrap';

/**
 * @param {object} props react props
 * @example
 <Filter name="Hello world"></Filter>
 */
function Filter({ onFilter }) {
	function onFilterHandler(event) {
		return onFilter(event.target.value, event);
	}

	function onSubmit(event) {
		event.preventDefault();
		return onFilter(this.value, event);
	}

	return (
		<form
			className="navbar-form navbar-right"
			role="search"
			onSubmit={onSubmit}
		>
			<div className="form-group">
				<div className="input-group">
					<input
						type="search"
						className="form-control"
						placeholder="Filter"
						aria-label="Filter"
						onChange={onFilterHandler}
						ref={c => onSubmit.bind(c)}
					/>
				</div>
				<Button
					bsStyle="link"
					type="submit"
					title="Filter"
				>
					<i className="fa fa-search" aria-hidden="true" />
					<span className="sr-only">Filter</span>
				</Button>
			</div>
		</form>
	);
}

Filter.propTypes = {
	onFilter: PropTypes.func,
};

export default Filter;
