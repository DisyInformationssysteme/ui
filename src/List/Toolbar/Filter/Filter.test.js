import React from 'react';
import renderer from 'react-test-renderer';

import Filter from './Filter.component';

jest.mock('react-dom');

describe('Filter', () => {
	it('should render its name', () => {
		const wrapper = renderer.create(<Filter name="Hello world" />).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});
