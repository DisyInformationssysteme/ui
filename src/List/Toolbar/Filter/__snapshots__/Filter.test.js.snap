exports[`Filter should render its name 1`] = `
<form
  className="navbar-form navbar-right"
  onSubmit={[Function]}
  role="search">
  <div
    className="form-group">
    <div
      className="input-group">
      <input
        aria-label="Filter"
        className="form-control"
        onChange={[Function]}
        placeholder="Filter"
        type="search" />
    </div>
    <button
      className="btn btn-link"
      disabled={false}
      title="Filter"
      type="submit">
      <i
        aria-hidden="true"
        className="fa fa-search" />
      <span
        className="sr-only">
        Filter
      </span>
    </button>
  </div>
</form>
`;
